parachains/runtimes/assets/asset-hub-westend/src/xcm_config.rs

```diff
@@ -469,12 +482,19 @@ impl xcm_executor::Config for XcmConfig {
 	>;
 	type Trader = (
 		UsingComponents<WeightToFee, WestendLocation, AccountId, Balances, ToStakingPot<Runtime>>,
-		cumulus_primitives_utility::TakeFirstAssetTrader<
-			AccountId,
-			AssetFeeAsExistentialDepositMultiplierFeeCharger,
```

Comment:
My understand is, that we can use the same `TakeFirstAssetTrader` and only  provide a different `BalanceToAssetBalance` converter for `AssetFeeAsExistentialDepositMultiplierFeeCharger` type. 
The converter which does the conversion based on the price provided by the pool, instead of min balances ration.
The rest can probably be reused.

One important thing is that the `Trader` is not supposed to perform any transfers/swaps, but only subtract the amount from the `payment` it receives as an argument, place it into its own registry to be paid later on `drop` and return the `unused`.
The `payment` received as an argument is a credit from the executor's holding register, which was already withdrawn from the user's account.
I am not really an expert in this topic, but this how I understanding it after reading the executor codebase and it make sense to me, please validate.